src/mscorlib/shared/System/IO/PathInternal.Unix.cs

```diff
@@ -55,23 +57,29 @@ internal static string NormalizeDirectorySeparators(string path)
                 }
             }
 
-            if (normalized) return path;
+            if (normalized)
+                return path;
 
-            StringBuilder builder = new StringBuilder(path.Length);
-
-            for (int i = 0; i < path.Length; i++)
+            fixed (char* f = path)
             {
-                char current = path[i];
+                return string.Create(path.Length, (Path: (IntPtr)f, PathLength: path.Length), (dst, state) =>
```

Comment:
This will create a string that is longer than what you want. I think that the loop above should run till end of the path to compute the require string length.